renderdoc/driver/d3d12/d3d12_debug.cpp

```diff
@@ -2916,10 +2916,14 @@ void D3D12Replay::PixelHistory::Init(WrappedID3D12Device *device, D3D12DebugMana
   shaderCache->GetShaderBlob(hlsl.c_str(), "RENDERDOC_PrimitiveIDPS",
                              D3DCOMPILE_WARNINGS_ARE_ERRORS, {}, "ps_6_0", &PrimitiveIDPSDxil);
 
-  shaderCache->GetShaderBlob(hlsl.c_str(), "RENDERDOC_PixelHistoryFixedColPS",
-                             D3DCOMPILE_WARNINGS_ARE_ERRORS, {}, "ps_5_0", &FixedColorPS);
-  shaderCache->GetShaderBlob(hlsl.c_str(), "RENDERDOC_PixelHistoryFixedColPS",
-                             D3DCOMPILE_WARNINGS_ARE_ERRORS, {}, "ps_6_0", &FixedColorPSDxil);
+  for(int i = 0; i < D3D12_SIMULTANEOUS_RENDER_TARGET_COUNT; ++i)
```

Comment:
Rather than separate shaders per RT index, would it be possible to use a uniform to specify which one to write to? It feels unfortunate to need to create 16 PSOs for this.

Reply:
I'm not aware of a way to have a single pixel shader that optionally writes to targets. It's an all or nothing proposition as far as I'm aware. Is there a feature of D3D12/HLSL I'm missing?

Reply:
I think it is 16 shader blobs rather than 16 explicitly created PSOs.
Also it is not uncommon to create a lot of PSOs during replay : the overlay for instance creates quite a lot to cope with different formats/shaders and sample count.

In theory (but I don't think it is required now) - the shaders could be created and cached on demand when the specific output index is known. I don't think need all eight during a single pixel history.

Reply:
Perhaps another thing to try would be having the fixed color shader usage fill the unbound RT indices with dummy targets to avoid the validation warning? Though this is not anything worth slowing down the PR for, imo.